src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java

```diff
@@ -254,6 +284,48 @@ else if (targetY <= levelDepth && workerY > levelDepth)
             //If he is on the same Y level as his target and both underground, don't use a proxy. Just don't.
             else if (targetY <= levelDepth)
             {
+                double closestNode = Double.MAX_VALUE;
+                Node lastNode = null;
+                for(Map.Entry<Point2D, Node> node : level.getNodes().entrySet())
+                {
+                    double distanceToNode = node.getKey().distance(worker.getPosition().getX(), worker.getPosition().getZ());
+                    if(distanceToNode < closestNode)
+                    {
+                        lastNode = node.getValue();
+                        closestNode = distanceToNode;
+                    }
+                }
+
+                if(lastNode != null && lastNode.getParent() != null)
+                {
+                    com.minecolonies.coremod.entity.ai.citizen.miner.Node currentNode = level.getNode(lastNode.getParent());
+                    while (new Point2D.Double(currentNode.getX(), currentNode.getZ()) != currentNode.getParent() && currentNode.getParent() != null)
+                    {
+                        proxyList.add(new BlockPos(currentNode.getX(), levelDepth, currentNode.getZ()));
+                        currentNode = level.getNode(currentNode.getParent());
+                    }
+                }
+
+                if(level.getRandomNode().getParent() != null)
+                {
+                    final List<BlockPos> nodesToTarget = new ArrayList<>();
+                    com.minecolonies.coremod.entity.ai.citizen.miner.Node currentNode = level.getNode(level.getRandomNode().getParent());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Unnecessary use of fully qualified name 'com.minecolonies.coremod.entity.ai.citizen.miner.Node' due to existing import 'com.minecolonies.coremod.entity.ai.citizen.miner.*' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AUnnecessaryFullyQualifiedName)